### PR TITLE
Feature/plat 317 plat plat 318 rbac setup

### PR DESCRIPTION
**Pull request by relopezz**
-  working schema

- also have migration script ready

- relationships are validated

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,21 +1,16 @@
-FROM golang:1.12-buster AS builder
-
-RUN     curl -sL -o /tmp/s6-overlay-amd64.tar.gz https://github.com/just-containers/s6-overlay/releases/download/v1.22.1.0/s6-overlay-amd64.tar.gz \
```
Comment (greggilbert) [inline Dockerfile]: Why was this whole `RUN` line removed?

File: domain/permission.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package domain
+
+import "time"
+
+// Permission : Permission Fields"
+type Permission struct {
+	ID          int       `json:"id,primary_key"`
```
Comment (greggilbert) [inline domain/permission.go]: @relopezz Take a look at http://gorm.io/docs/conventions.html - you can add `gorm.Model` into the Structs and it'll create ID, CreatedAt, UpdatedAt, and DeletedAt. Easier than writing it out a bunch of times in all the models.

File: domain/rolepermission.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package domain
+
+import "time"
+
+// RolePermission : Releation between Roles and Permissions"
+type RolePermission struct {
```
Comment (greggilbert) [inline domain/rolepermission.go]: Why does this table need the created/updated/deleted?

Comment (relopezz) [reply domain/rolepermission.go]: when you assign/remove a permission to a role?

File: domain/permission.go (hunk 2 above)
Comment (relopezz) [reply domain/permission.go]: I updated, thanks

**Pull request merged**